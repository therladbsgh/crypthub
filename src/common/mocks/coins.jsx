export const coins = [
    {
        name: 'Bitcoin',
        symbol: 'BTC',
        currPrice: 10000
    },
    {
        name: 'Ethereum',
        symbol: 'ETH',
        currPrice: 5000
    }
]
