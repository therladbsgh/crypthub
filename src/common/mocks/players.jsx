import date from 'date-and-time';

export const player1 = {
    name: 'player1',
    portfolio: [
        {
            name: 'Bitcoin',
            amount: 250
        },
        {
            name: 'Ethereum',
            amount: 1000.28373623
        },
        {
            name: 'USD',
            amount: 250
        }
    ],
    transactions: {
        history: [
            {
                id: '1',
                type: 'market',
                side: 'buy',
                size: 25,
                price: 100,
                symbol: 'BTC',
                date: date.parse('Apr 1 2018', 'MMM D YYYY'),
                GTC: true,
                expiration: null,
                filled: true,
                filledDate: date.parse('Apr 3 2018', 'MMM D YYYY')
            },
            {
                id: '2',
                type: 'limit',
                side: 'buy',
                size: 35.000006,
                price: 200,
                symbol: 'ETH',
                date: date.parse('Apr 2 2018', 'MMM D YYYY'),
                GTC: false,
                expiration: date.parse('Apr 5 2018', 'MMM D YYYY'),
                filled: false,
                filledDate: null
            }
        ],
        current: [
            {
                id: '3',
                type: 'short',
                side: 'sell',
                size: 80,
                price: 100,
                symbol: 'BTC',
                date: date.parse('Apr 2 2018', 'MMM D YYYY'),
                GTC: true,
                expiration: null,
                filled: false,
                filledDate: null
            },
            {
                id: '4',
                type: 'limit',
                side: 'buy',
                size: 2,
                price: 200,
                symbol: 'ETH',
                date: date.parse('Apr 7 2018', 'MMM D YYYY'),
                GTC: false,
                expiration: date.parse('May 2 2018', 'MMM D YYYY'),
                filled: false,
                filledDate: null
            }
        ]
    },
    netWorth: 200,
    numTrades: 80,
    netReturn: -20,
    todayReturn: 5,
    currRank: 2,
    buyingPower: 150,
    shortReserve: 50
};

export const player2 = {
    name: 'player2',
    portfolio: [
        {
            name: 'Bitcoin',
            amount: 300
        }
    ],
    transactions: {
        history: [
            {
                id: '5',
                side: 'buy',
                size: 1,
                price: 10,
                symbol: 'ETH'
            },
            {
                id: '6',
                side: 'sell',
                size: 3500,
                price: 202220,
                symbol: 'ETH'
            }
        ],
        current: [
            {
                id: '7',
                side: 'sell',
                size: 23230,
                price: 232,
                symbol: 'BTC'
            },
            {
                id: '8',
                side: 'buy',
                size: 6000,
                price: 50,
                symbol: 'ETH'
            }
        ]
    },
    netWorth: 300,
    numTrades: 2,
    netReturn: 120
};
