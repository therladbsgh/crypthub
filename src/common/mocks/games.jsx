import date from 'date-and-time';
import * as PlayerMocks from './players';

export const game1 = {
    id: '1',
    name: 'game1',
    description: 'desc1',
    host: 'user1',
    created: date.parse('Apr 1 2018', 'MMM D YYYY'),
    start: date.parse('Apr 2 2018', 'MMM D YYYY'),
    end: date.parse('May 7 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};

export const game2 = {
    id: '2',
    name: 'game2',
    description: 'desc2',
    host: 'user2',
    created: date.parse('Apr 2 2018', 'MMM D YYYY'),
    start: date.parse('Apr 3 2018', 'MMM D YYYY'),
    end: date.parse('May 1 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};

export const game3 = {
    id: '3',
    name: 'game3',
    description: 'desc3',
    host: 'user3',
    created: date.parse('Apr 4 2018', 'MMM D YYYY'),
    start: date.parse('Apr 8 2018', 'MMM D YYYY'),
    end: date.parse('Apr 28 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};
