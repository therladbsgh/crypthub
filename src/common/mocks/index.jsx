import * as GameMocks from './games';
import * as PlayerMocks from './players';

export { GameMocks, PlayerMocks };
