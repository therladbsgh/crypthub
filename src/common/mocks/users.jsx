import * as GameMocks from './games';

export const user1 = {
    id: 'id1',
    name: 'user1',
    ELO: 3000,
    games: [
        GameMocks.game1,
        GameMocks.game2
    ]
};

export const user2 = {
    id: 'id2',
    name: 'user2',
    ELO: 2000,
    games: [
        GameMocks.game2,
        GameMocks.game3
    ]
};

export const user3 = {
    id: 'id3',
    name: 'user3',
    ELO: 2000,
    games: [
        GameMocks.game2,
        GameMocks.game3
    ]
};
