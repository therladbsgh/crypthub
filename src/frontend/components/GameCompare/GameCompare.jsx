import * as _ from 'lodash';
import React, { Component } from 'react';
import { Header, Dropdown } from 'semantic-ui-react';
import { Portfolio } from 'components';

export default class GameCompare extends Component {
	constructor(props) {
		super(props);

		this.state = {
			playerIds: _.map(this.props.players, '_id'),
		};

		this.handleChange = this.handleChange.bind(this);
	}

	handleChange(event, { value }) {
		this.setState({ playerIds: value });
	}
	
	render() {
        const { players, completed } = this.props;
        const { playerIds } = this.state;

		return (
			<div>
                <Header as='h2'>Compare Player Portfolios</Header>
				<Dropdown placeholder='Players to compare' multiple search selection options={_.map(players, p => ({ text: p.name, value: p.id }))} value={playerIds} onChange={this.handleChange} />
                {_.map(playerIds, _id => {
                    const player = _.find(players, { _id });
                    return (
                        <Portfolio key={_id} portfolio={player.portfolio} header={player.name} completed={completed} />
                    );
                })}
			</div>
		);
	}
}
