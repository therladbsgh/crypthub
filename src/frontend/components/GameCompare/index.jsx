import GameCompare from './GameCompare';

export default GameCompare;
