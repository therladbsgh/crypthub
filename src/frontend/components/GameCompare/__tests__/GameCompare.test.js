import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import GameCompare from '../GameCompare';
import { GameMocks } from 'mocks';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  	const tree = renderer.create(<GameCompare players={GameMocks.game1.players} />).toJSON();
  	expect(tree).toMatchSnapshot();
});