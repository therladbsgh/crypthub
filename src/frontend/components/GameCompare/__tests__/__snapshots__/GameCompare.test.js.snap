// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2
    className="ui header"
  >
    Compare Player Portfolios
  </h2>
  <div
    aria-busy={undefined}
    aria-disabled={undefined}
    aria-expanded={false}
    className="ui multiple search selection dropdown"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="combobox"
    tabIndex={undefined}
  >
    <input
      aria-autocomplete="list"
      autoComplete="off"
      className="search"
      onChange={[Function]}
      style={
        Object {
          "width": undefined,
        }
      }
      tabIndex={0}
      type="text"
      value=""
    />
    <span
      className="sizer"
    />
    <div
      aria-live="polite"
      className="text"
      role="alert"
    />
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      aria-multiselectable={true}
      className="menu transition"
      role="listbox"
    >
      <div
        className="message"
      >
        No results found.
      </div>
    </div>
  </div>
  <table
    className="ui celled table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <th
          className=""
        >
          Symbol
        </th>
        <th
          className=""
        >
          % of Holdings
        </th>
        <th
          className=""
        >
          Amount
        </th>
        <th
          className=""
        >
          Current Price (Today's Change)
        </th>
        <th
          className=""
        >
          Current Value
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className="positive"
      >
        <td
          className=""
        >
          BTC
        </td>
        <td
          className=""
        >
          29.41%
        </td>
        <td
          className=""
        >
          250.00000000
        </td>
        <td
          className=""
        >
          $10,000.00 (+3.25%)
        </td>
        <td
          className=""
        >
          $2,500,000.00
        </td>
      </tr>
      <tr
        className="error"
      >
        <td
          className=""
        >
          ETH
        </td>
        <td
          className=""
        >
          58.83%
        </td>
        <td
          className=""
        >
          1,000.28373623
        </td>
        <td
          className=""
        >
          $5,000.00 (-20.01%)
        </td>
        <td
          className=""
        >
          $5,001,418.68
        </td>
      </tr>
      <tr
        className=""
      >
        <td
          className=""
        >
          USD
        </td>
        <td
          className=""
        >
          11.76%
        </td>
        <td
          className=""
        >
          $1,000,000.00
        </td>
        <td
          className=""
        >
          -
        </td>
        <td
          className=""
        >
          $1,000,000.00
        </td>
      </tr>
    </tbody>
    <tfoot
      className=""
    >
      <tr
        className=""
      >
        <th
          className=""
        >
          <strong>
            TOTAL
          </strong>
        </th>
        <th
          className=""
        >
          <strong>
            100.00%
          </strong>
        </th>
        <th
          className=""
        >
          <strong>
            -
          </strong>
        </th>
        <th
          className=""
        >
          <strong>
            -
          </strong>
        </th>
        <th
          className=""
        >
          <strong>
            $8,501,418.68
          </strong>
        </th>
      </tr>
    </tfoot>
  </table>
  <table
    className="ui celled table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <th
          className=""
        >
          Symbol
        </th>
        <th
          className=""
        >
          % of Holdings
        </th>
        <th
          className=""
        >
          Amount
        </th>
        <th
          className=""
        >
          Current Price (Today's Change)
        </th>
        <th
          className=""
        >
          Current Value
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className="positive"
      >
        <td
          className=""
        >
          BTC
        </td>
        <td
          className=""
        >
          28.21%
        </td>
        <td
          className=""
        >
          200.42500000
        </td>
        <td
          className=""
        >
          $10,000.00 (+3.25%)
        </td>
        <td
          className=""
        >
          $2,004,250.00
        </td>
      </tr>
      <tr
        className="error"
      >
        <td
          className=""
        >
          ETH
        </td>
        <td
          className=""
        >
          54.90%
        </td>
        <td
          className=""
        >
          780.00000002
        </td>
        <td
          className=""
        >
          $5,000.00 (-20.01%)
        </td>
        <td
          className=""
        >
          $3,900,000.00
        </td>
      </tr>
      <tr
        className=""
      >
        <td
          className=""
        >
          USD
        </td>
        <td
          className=""
        >
          16.89%
        </td>
        <td
          className=""
        >
          $1,200,000.00
        </td>
        <td
          className=""
        >
          -
        </td>
        <td
          className=""
        >
          $1,200,000.00
        </td>
      </tr>
    </tbody>
    <tfoot
      className=""
    >
      <tr
        className=""
      >
        <th
          className=""
        >
          <strong>
            TOTAL
          </strong>
        </th>
        <th
          className=""
        >
          <strong>
            100.00%
          </strong>
        </th>
        <th
          className=""
        >
          <strong>
            -
          </strong>
        </th>
        <th
          className=""
        >
          <strong>
            -
          </strong>
        </th>
        <th
          className=""
        >
          <strong>
            $7,104,250.00
          </strong>
        </th>
      </tr>
    </tfoot>
  </table>
</div>
`;
